### PR TITLE
Fixing Dynamic Mode

### PR DESCRIPTION
**Pull request by DeityLink**
gentlemen, we got im

:cl:
* bugfix: Dynamic Mode: fixed High Population Override counting non-ready players toward its trigger.
* bugfix: Dynamic Mode: fixed basically no ruleset other than traitor or vampire firing at roundstart for the past two weeks.

### CODE REVIEW

**Comment by DeityLink** [comment]
wait hold on

**Comment by DeityLink** [comment]
Alright

File: code/datums/gamemode/dynamic/dynamic_rulesets.dm [hunk 1]
```diff
@@ -237,6 +237,13 @@
 
 /datum/dynamic_ruleset/roundstart/ready(var/forced = 0)
 	if (!forced)
-		if(!check_enemy_jobs())
+		var/job_check = 0
```
Comment (DamianX) [inline code/datums/gamemode/dynamic/dynamic_rulesets.dm]: Can you override check_enemy_jobs instead? It's the whole point.

Comment (DeityLink) [reply code/datums/gamemode/dynamic/dynamic_rulesets.dm]: fine

**Comment by ShiftyRail** [comment]
Why would this bug manifest itself only for the past two weeks, if this part of the code hadn't been touched in months

**Comment by DeityLink** [comment]
@ShiftyRail Except it was, see #23432 as the merge date also corresponds perfectly with the bug starting to happen, and the code change makes complete sense for it to be broken.

I guess the "0% tested tag" was justified

**Pull request merged**